sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -2328,4 +2327,27 @@ public Response<BlobLegalHoldResult> setLegalHoldWithResponse(boolean legalHold,
 
         return blockWithOptionalTimeout(response, timeout);
     }
+
+    private static final class ByteBufferBackedOutputStream extends OutputStream {
```

Comment:
This looks to be duplicated in a few spots, let's make this a class somewhere in implementation so it can be reused.